contributors/design-proposals/no-new-privs.md

```diff
@@ -0,0 +1,54 @@
+#Support "no new privileges" in Kubernetes
+
+##Description
+
+In Linux, the `execve` system call can grant more privileges to a newly-created process than its parent process. Considering security issues, since Linux kernel v3.5, there is a new flag named `no_new_privs` added to prevent those new privileges from being granted to the processes. 
+
+`no_new_privs` is inherited across `fork`, `clone` and `execve` and can not be unset. With `no_new_privs` set, `execve` promises not to grant the privilege to do anything that could not have been done without the `execve` call.
+
+Docker started to support `no_new_privs` option since 1.11. Here is the [link](https://github.com/docker/docker/issues/20329) of the ticket in Docker community to support `no_new_privs` option.
+
+We want to support the creation of containers with `no_new_privs` enabled in Kubernetes, which will make the Kubernetes cluster more safe. Here is the [link](https://github.com/kubernetes/kubernetes/issues/38417) of the ticket in Kubernetes community to track this proposal.
+
+
+##Current implementation
+
+###Support in Docker
```

Comment:
rkt has an isolator for it, but does not expose it as a CLI flag at this time.

It's a kinda low-level knob and asking the user to tweak it isn't great.

To change it in rkt, an image you package can indicate whether it should be run this way or not (e.g. when I package nginx I might assert it can run with no-new-privs and rkt will apply it, but when I package my irc bot which execs suid ping I wouldn't set the isolator and rkt wouldn't apply it).

It wouldn't be too difficult to expose it to the user or to, from rktlet.